builtin/providers/aws/resource_aws_elb.go

```diff
@@ -725,44 +725,61 @@ func resourceAwsElbUpdate(d *schema.ResourceData, meta interface{}) error {
 
 		removed := expandStringList(os.Difference(ns).List())
 		added := expandStringList(ns.Difference(os).List())
+		subnetconn := meta.(*AWSClient).ec2conn
 
-		if len(removed) > 0 {
-			detachOpts := &elb.DetachLoadBalancerFromSubnetsInput{
-				LoadBalancerName: aws.String(d.Id()),
-				Subnets:          removed,
+		// This section sorts the removed/added subnets based on availability zones
+		dup_az_count := 0
+		for i, r_subnet_id := range removed {
+			r_subnet, err := subnetconn.DescribeSubnets(&ec2.DescribeSubnetsInput{
+				SubnetIds: []*string{r_subnet_id},
+			})
+			if err != nil {
+				return err
+			}
+			for j, a_subnet_id := range added {
+				a_subnet, err := subnetconn.DescribeSubnets(&ec2.DescribeSubnetsInput{
```

Comment:
This looks like it will make an API call for every subnet in the added subnet list for every subnet in the removed subnet list. So we'll be making `len(added) * len(removed)` API calls, which seems excessive. Couldn't we pull down each once and no go combinatorial on it? I don't think the values will change, right?

Also, could we replace this entire block with a call to [`sort.Slice`](https://golang.org/pkg/sort#Slice)?